ownCloud/UI Elements/RoundedInfoView.swift

```diff
@@ -0,0 +1,94 @@
+//
+//  RoundedInfoView.swift
+//  ownCloud
+//
+//  Created by Matthias Hühne on 19.03.19.
+//  Copyright © 2019 ownCloud GmbH. All rights reserved.
+//
+
+/*
+* Copyright (C) 2019, ownCloud GmbH.
+*
+* This code is covered by the GNU Public License Version 3.
+*
+* For distribution utilizing Apple mechanisms please see https://owncloud.org/contribute/iOS-license-exception/
+* You should have received a copy of this license along with this program. If not, see <http://www.gnu.org/licenses/gpl-3.0.en.html>.
+*
+*/
+
+import UIKit
+
+class RoundedInfoView: UIView, Themeable {
+
+	// MARK: - Constants
+	let cornerRadius : CGFloat = 10.0
+	let borderWidth : CGFloat = 1.0
+	let horizontalPadding : CGFloat = 10.0
+	let verticalPadding : CGFloat = 20.0
+	let verticalLabelPadding : CGFloat = 10.0
+	let font : UIFont = UIFont.systemFont(ofSize: 14)
+
+	// MARK: - Instance Variables
+	var infoText : String = ""
+	var messageThemeApplierToken : ThemeApplierToken?
+	var backgroundView = UIView()
+	var label = UILabel()
+
+	init(text: String) {
+		super.init(frame: CGRect.zero)
+		infoText = text
+		Theme.shared.register(client: self, applyImmediately: true)
+		styleView()
+	}
+
+	required init?(coder aDecoder: NSCoder) {
+		fatalError("init(coder:) has not been implemented")
+	}
+
+	deinit {
+		Theme.shared.unregister(client: self)
+
+		if messageThemeApplierToken != nil {
+			Theme.shared.remove(applierForToken: messageThemeApplierToken)
+			messageThemeApplierToken = nil
+		}
+	}
+
+	private func styleView() {
```

Comment:
@hosy may be `styleView` doesn't fully reflect what it does.. since it is also laying out the view

Reply:
Solved